pkg/resources/grant_ownership_test.go

```diff
@@ -0,0 +1,518 @@
+			Identifier: GrantOwnershipId{
+				Kind: OnFutureGrantOwnershipKind,
+				Data: &BulkOperationGrantData{
+					ObjectNamePlural: sdk.PluralObjectTypeTables,
+					Kind:             InSchemaBulkOperationGrantKind,
+					Schema:           sdk.Pointer(sdk.NewDatabaseObjectIdentifier("test_database", "test_schema")),
+				},
+			},
+			ExpectedShowGrantsOpts: &sdk.ShowGrantOptions{
+				Future: sdk.Bool(true),
+				In: &sdk.ShowGrantsIn{
+					Schema: sdk.Pointer(sdk.NewDatabaseObjectIdentifier("test_database", "test_schema")),
+				},
+			},
+			ExpectedGrantedOn: sdk.ObjectTypeTable,
+		},
+	}
+
+	for _, tt := range testCases {
+		tt := tt
+		t.Run(tt.Name, func(t *testing.T) {
+			opts, grantedOn := prepareShowGrantsRequestForGrantOwnership(&tt.Identifier)
+			if tt.ExpectedShowGrantsOpts == nil {
+				assert.Nil(t, opts)
+			} else {
+				assert.NotNil(t, opts)
+				assert.Equal(t, *tt.ExpectedShowGrantsOpts, *opts)
+			}
+			assert.Equal(t, tt.ExpectedGrantedOn, grantedOn)
+		})
+	}
+}
+
+func TestGetOwnershipGrantTo(t *testing.T) {
+	testCases := []struct {
+		Name         string
+		AccountRole  *string
+		DatabaseRole *string
+		Expected     sdk.OwnershipGrantTo
+		ExpectPanic  bool
+	}{
+		{
+			Name:        "account role name",
+			AccountRole: sdk.String("account_role_name"),
+			Expected: sdk.OwnershipGrantTo{
+				AccountRoleName: sdk.Pointer(sdk.NewAccountObjectIdentifier("account_role_name")),
+			},
+		},
+		{
+			Name:        "account role name - quoted",
+			AccountRole: sdk.String("\"account_role_name\""),
+			Expected: sdk.OwnershipGrantTo{
+				AccountRoleName: sdk.Pointer(sdk.NewAccountObjectIdentifier("account_role_name")),
+			},
+		},
+		{
+			Name:         "database role name",
+			DatabaseRole: sdk.String("test_database.database_role_name"),
+			Expected: sdk.OwnershipGrantTo{
+				DatabaseRoleName: sdk.Pointer(sdk.NewDatabaseObjectIdentifier("test_database", "database_role_name")),
+			},
+		},
+		{
+			Name:         "database role name - quoted",
+			DatabaseRole: sdk.String("\"test_database\".\"database_role_name\""),
+			Expected: sdk.OwnershipGrantTo{
+				DatabaseRoleName: sdk.Pointer(sdk.NewDatabaseObjectIdentifier("test_database", "database_role_name")),
+			},
+		},
+		{
+			Name:        "validation - incorrect account role name",
+			AccountRole: sdk.String("database_name.account_role_name"),
+			ExpectPanic: true,
+		},
+		{
+			Name:         "validation - incorrect database role name",
+			DatabaseRole: sdk.String("database_name.schema_name.database_role_name"),
+			ExpectPanic:  true,
+		},
+	}
+
+	for _, tt := range testCases {
+		tt := tt
+		t.Run(tt.Name, func(t *testing.T) {
+			config := make(map[string]any)
+			if tt.AccountRole != nil {
+				config["account_role_name"] = *tt.AccountRole
+			}
+			if tt.DatabaseRole != nil {
+				config["database_role_name"] = *tt.DatabaseRole
+			}
+			d := schema.TestResourceDataRaw(t, grantOwnershipSchema, config)
+
+			defer func() {
```

Comment:
parametrized tests are nice but if the setup becomes like this, I think it's more readable (and less error-prone) to setup to set of tests: one for valid and second for invalid configs. In this case for invalid ones we could just use assert/require.Panics(). Also, there is clear separation between db and account roles, so I think that splitting it into 3 different sets would be best, the ost readable option

Reply:
Divided based on valid / invalid and role type